src/github.com/matrix-org/go-neb/clients/clients.go

```diff
@@ -343,35 +319,49 @@ func (c *Clients) onRoomMemberEvent(client *matrix.Client, event *matrix.Event)
 	}
 }
 
-func (c *Clients) newClient(config api.ClientConfig) (*matrix.Client, error) {
-	homeserverURL, err := url.Parse(config.HomeserverURL)
+func (c *Clients) newClient(config api.ClientConfig) (*gomatrix.Client, error) {
+	client, err := gomatrix.NewClient(config.HomeserverURL, config.UserID, config.AccessToken)
 	if err != nil {
 		return nil, err
 	}
-
-	client := matrix.NewClient(c.httpClient, homeserverURL, config.AccessToken, config.UserID)
-	client.NextBatchStorer = nextBatchStore{c.db}
-	client.ClientConfig = config
+	client.Client = c.httpClient
+	syncer := client.Syncer.(*gomatrix.DefaultSyncer)
+	client.Store = &matrix.NEBStore{
+		InMemoryStore: *gomatrix.NewInMemoryStore(),
+		Database:      c.db,
+		ClientConfig:  config,
+	}
+	syncer.Store = client.Store
 
 	// TODO: Check that the access token is valid for the userID by peforming
 	// a request against the server.
 
-	client.Worker.OnEventType("m.room.message", func(event *matrix.Event) {
+	syncer.OnEventType("m.room.message", func(event *gomatrix.Event) {
 		c.onMessageEvent(client, event)
 	})
 
-	client.Worker.OnEventType("m.room.bot.options", func(event *matrix.Event) {
+	syncer.OnEventType("m.room.bot.options", func(event *gomatrix.Event) {
 		c.onBotOptionsEvent(client, event)
 	})
 
 	if config.AutoJoinRooms {
-		client.Worker.OnEventType("m.room.member", func(event *matrix.Event) {
+		syncer.OnEventType("m.room.member", func(event *gomatrix.Event) {
 			c.onRoomMemberEvent(client, event)
 		})
 	}
 
 	if config.Sync {
-		go client.Sync()
+		go func() {
+			for {
+				if e := client.Sync(); e != nil {
```

Comment:
No, well spotted. When `StopSync()` is called there is a `nil` error which should result in it stopping then.